_overviews/tutorials/binary-compatibility-for-library-authors.md

```diff
@@ -178,11 +178,13 @@ Again, we recommend using MiMa to double-check that you have not broken binary c
 Sometimes, it is desirable to change the definition of a case class (adding and/or removing fields) while still staying backwards-compatible with the existing usage of the case class, i.e. not breaking the so-called _binary compatibility_. The first question you should ask yourself is “do you need a _case_ class?” (as opposed to a regular class, which can be easier to evolve in a binary compatible way). A good reason for using a case class is when you need a structural implementation of `equals` and `hashCode`.
 
 To achieve that, follow this pattern:
- * make the primary constructor private (this makes the `copy` method of the class private as well)
- * define a private `unapply` function in the companion object (note that by doing that the case class loses the ability to be used as an extractor in match expressions)
+ * make the primary constructor private (for Scala 3, this makes the `copy` method of the class and `apply` in the companion object private as well)
  * for all the fields, define `withXXX` methods on the case class that create a new instance with the respective field changed (you can use the private `copy` method to implement them)
  * create a public constructor by defining an `apply` method in the companion object (it can use the private constructor)
- * in Scala 2, you have to add the compiler option `-Xsource:3`
+ * (Scala 2) you have to add the compiler option `-Xsource:3`
+ * (Scala 2) define the `copy` method with all the current fields manually and set it as `private`
+ * (Scala 2) define a private `unapply` method in the companion object (note that by doing that the case class loses the ability to be used as an extractor in match expressions)
+ * (Scala 3) define a custom `fromProduct` method in the companion object
```

Comment:
That one is interesting. We need to investigate further:
- what is the current state of mirrors (without defining `fromProduct`)
- demonstrate in a project example that defining `fromProduct` is enough to get type class derivation working on case classes with a private primary constructor, and that it works as expected when fields are added.

Reply:
That discussion needs to be resolved before we can advertise implementing `fromProduct`.